### PR TITLE
[SYCL] Clean up tests moved to llvm-test-suite

### PR DESCRIPTION
**Pull request by vladimirlaz**
- remove sycl/test/esimd directory which is empty now;
 - remove inline assembler tests;
 - add sanity inline assembler test sycl/test/extensions/inline_asm.cpp.

### CODE REVIEW

File: sycl/test/extensions/inline_asm.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// UNSUPPORTED: cuda
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
```
Comment (bader) [inline sycl/test/extensions/inline_asm.cpp:3]: My understanding is that we are moving tests to llvm-test-suite, which require additional (to vanilla llvm) HW or SW dependencies. Following this logic this test should be moved to llvm-test-suite as well. Am I missing anything?

Tagging @andykaylor.

Comment (vladimirlaz) [inline sycl/test/extensions/inline_asm.cpp:3]: Remove execution on GPU device. Left compilation part.
```suggestion
```

Comment (vladimirlaz) [reply sycl/test/extensions/inline_asm.cpp]: I have removed execution on GPU device. So the test verifies that compilation succeeded

File: sycl/test/extensions/inline_asm.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
+// UNSUPPORTED: cuda
```
Comment (bader) [inline sycl/test/extensions/inline_asm.cpp:1]:
```suggestion

```

File: sycl/test/extensions/inline_asm.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
+// RUN: %clangxx -fsycl %s -o %t.out
```
Comment (AlexeySachkov) [inline sycl/test/extensions/inline_asm.cpp:1]:
```suggestion
// This is a simple basic acceptance test for inline ASM feature
// More test cases can be found in https://github.com/intel/llvm-test-suite/tree/intel/SYCL/InlineAsm
// RUN: %clangxx -fsycl %s -o %t.out
```

**Review by AlexeySachkov — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**